### ISSUE TITLE
Stateful LM behavior

### ISSUE THREAD

**Issue by lakshmi-speak** [kensho-technologies/pyctcdecode#66]
Is the `BeamSearchDecoderCTC` default behavior to utilize the state in the language model?
What does `DEFAULT_SCORE_LM_BOUNDARY = True` mean?

**Comment by lopez86** [comment]
kenlm language models include beginning of sentence and end of sentence characters in score. `DEFAULT_SCORE_LM_BOUNDARY = True` means that the default behavior is to begin the scoring with the beginning of sentence character, otherwise that is not added. See https://github.com/kpu/kenlm/blob/master/python/example.py for an example of how the language model works. For streaming, you can pass a state into the  BeamSearchDecoderCTC.decode() function

**Comment by dertilo** [comment]
to be honest I cannot see how streaming could be done currently, it seems that each time one calls some decoding-method it  starts at an empty [EMPTY_START_BEAM](https://github.com/kensho-technologies/pyctcdecode/blob/33478761427b3faad2652ca5b46b158566d88bab/pyctcdecode/decoder.py#L370) but actually it should take such a "start-beam" as input-argument, or not? So my understanding is that one needs 2 states for streaming, the LM-state and the beams-themselves, so that one can incrementally decode (chunks of logits) and keep growing the beams instead of starting from "scratch" (EMPTY_START_BEAM). 
Or am I understanding it wrongly?
